### ISSUE TITLE
Smoke Github Action 

### ISSUE THREAD

**Issue by jonaprieto** [SamirTalwar/Smoke#92]
Many projects living on Github could get the benefit of Smoke for testing in the CIs if there is an easy Smoke Github action. Maybe a starting point is: https://github.com/rwe/actions-hlint-run

https://docs.github.com/en/actions/creating-actions.

**Comment by SamirTalwar** [comment]
Looks good; please feel free to submit a PR.

**Comment by SamirTalwar** [comment]
Yeah, I'd like to do that. However, the current code on `main` introduces a dependency on ICU4C, and I need to make sure that this doesn't break everything on various setups, so it requires more manual testing than usual. (If it does, I will need to figure out another approach to regular expressions.)
